client/src/modules/stock/entry/modals/lots.modal.html

```diff
@@ -105,8 +121,8 @@
       FORM.BUTTONS.CLOSE
     </button>
 
-    <button type="submit" class="btn btn-primary" ng-click="$ctrl.submit(FindForm, $event)" data-method="submit" translate>
-      FORM.BUTTONS.SUBMIT
-    </button>
+    <bh-loading-button loading-state="$ctrl.$loading">
```

Comment:
This needs to be `FindForm.$loading`.  The loading is set on the form.

Reply:
How can I return the Promise.all(promises) and still execute the saveSetting(),  Instance.close() afterward?

Reply:
My language was imprecise. You need to return the promise chain that starts with Promise.all(). Just out a return statement in from of it.

Reply:
I'm obviously missing something here!  I tried that but VS Code complained.  See below
![image](https://user-images.githubusercontent.com/62145/123516706-618c7c00-d695-11eb-93e8-5f228426074c.png)

Are you suggesting that I can get rid of the .then() afterwards?

Reply:
No, the error here is because you have mixed `return` values.  It's a style thing, not a real error.

There is an earlier `return` above that returns nothing during an `if` check.  Just add `return 0;` to it, and ESLint will shut up.